snowdrop-block/src/Snowdrop/Block/Application.hs

```diff
@@ -229,16 +233,15 @@ applyFork :: forall blkType m conf txtypes xs
     -> m (OldestFirst [] (ChgAccum conf))
 applyFork cfg fork ForkApplyAction{..} = do
     toApply <- txsToApply -- Apply tx1, ..., tx_k to state
-    forkUndoss <- liftERoComp $ traverse (traverse (computeUndo @xs @conf)) chgAccums
-    let forkUndos = join forkUndoss
-    let forkBlockUndos = gett <$> forkUndos
-    (liftERoComp $ modifyAccumUndo @xs @conf (NewestFirst forkUndos)) >>= modify . flip sett
 
+    -- 7.Compute undos for blocks in fork (also using acc0, .., acc_{i-1}).
+    forkUndos <- liftERoComp $ traverse (computeUndo @xs @conf) chgAccum
+    let forkBlockUndos = gett <$> forkUndos
```

Comment:
I see that tip is changed in `setTip @blkType faaRollbackTo`, but actual rollback of state didn't happen.

One needs to iterate over the blunds and for each blund apply undo and remove blund from storage.